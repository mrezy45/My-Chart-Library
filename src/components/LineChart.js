import React from 'react';
import '../styles/ChartStyles.css';

const LineChart = ({ data, colors = [] }) => {
  const maxValue = Math.max(...data.map(d => d.value));
  const svgHeight = 300;
  const points = data
    .map((d, i) => `${i * 50},${svgHeight - (d.value / maxValue) * svgHeight}`)
    .join(' ');

  return (
    <div className="chart-container">
      <svg width="100%" height="300px">
        <polyline
          fill="none"
          stroke={colors[0] || 'steelblue'}
          strokeWidth="2"
          points={points}
        />
      </svg>
    </div>
  );
};

export default LineChart;