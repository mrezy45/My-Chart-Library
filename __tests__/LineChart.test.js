import React from 'react';
import { render } from '@testing-library/react';
import LineChart from '../src/components/LineChart';

describe('LineChart Component', () => {
  it('renders without crashing', () => {
    const data = [{ value: 50 }, { value: 80 }];
    const { container } = render(<LineChart data={data} colors={['#ff0000']} />);
    expect(container).toMatchSnapshot();
  });

  it('renders the polyline correctly', () => {
    const data = [{ value: 50 }, { value: 80 }];
    const { container } = render(<LineChart data={data} colors={['#ff0000']} />);
    const polyline = container.querySelector('polyline');
    expect(polyline).toBeInTheDocument();
  });
});