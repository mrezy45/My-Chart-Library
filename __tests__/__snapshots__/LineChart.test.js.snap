// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart Component renders without crashing 1`] = `
<div>
  <div
    class="chart-container"
  >
    <svg
      height="300px"
      width="100%"
    >
      <polyline
        fill="none"
        points="0,112.5 50,0"
        stroke="#ff0000"
        stroke-width="2"
      />
    </svg>
  </div>
</div>
`;
